Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,31 @@
+# create a lock using `flock (2)`. A name is required as first argument.
+# the lock will be automatically unlocked when the shell process quits.
+lock() {
+  local name="$1"
+  local lock_file="/tmp/homebrew-${name}.lock"
+  # open the lock file to file descriptor 3, so the shell process can hold the lock.
```

Comment:
What if another script is running and also using `3`? Regardless, the justification should be documented and `3` pulled into a variable where possible.